soroban-token-sdk/src/event.rs

```diff
@@ -17,7 +23,12 @@ impl Events {
             .publish(topics, (amount, expiration_ledger));
     }
 
-    pub fn transfer(&self, from: Address, to: Address, amount: i128) {
+    pub fn transfer<AddressType: sealed::IsAddressType + IntoVal<Env, Val>>(
+        &self,
+        from: AddressType,
+        to: AddressType,
+        amount: i128,
+    ) {
         let topics = (symbol_short!("transfer"), from, to);
         self.env.events().publish(topics, amount);
```

Comment:
Looks like the muxed address is emitted in the topic, so it wouldn't be very easy for a consumer to filter only on the underlying address.